usecases/consumption_test.go

```diff
@@ -0,0 +1,348 @@
+		func() *http.Response {
+			return &http.Response{
+				Status:     "200 OK",
+				StatusCode: 200,
+				Header:     http.Header{"Content-Type": []string{"application/json"}},
+				Body:       io.NopCloser(strings.NewReader(string("{\n  \"value\": 0,\n  \"unit\": \"\",\n  \"timestamp\": \"0001-01-01T00:00:00Z\",\n  \"version\": \"SignalA_v1.0\"\n}"))),
+			}
+		},
+		1,
+		errHTTP,
+		nil,
+		errHTTP,
+	},
+	{
+		"Status code error",
+		newTestCerviceWithNodes(),
+		&testSys,
+		[]byte("{\n  \"value\": 0,\n  \"unit\": \"\",\n  \"timestamp\": \"0001-01-01T00:00:00Z\",\n  \"version\": \"SignalA_v1.0\"\n}"),
+		func() *http.Response {
+			return &http.Response{
+				Status:     "300 NAK",
+				StatusCode: 300,
+				Header:     http.Header{"Content-Type": []string{"application/json"}},
+				Body:       io.NopCloser(strings.NewReader(string("{\n  \"value\": 0,\n  \"unit\": \"\",\n  \"timestamp\": \"0001-01-01T00:00:00Z\",\n  \"version\": \"SignalA_v1.0\"\n}"))),
+			}
+		},
+		2,
+		errHTTP,
+		nil,
+		errHTTP,
+	},
+	{
+		"io.ReadAll() error",
+		newTestCerviceWithNodes(),
+		&testSys,
+		[]byte("{\n  \"value\": 0,\n  \"unit\": \"\",\n  \"timestamp\": \"0001-01-01T00:00:00Z\",\n  \"version\": \"SignalA_v1.0\"\n}"),
+		func() *http.Response {
+			return &http.Response{
+				Status:     "200 OK",
+				StatusCode: 200,
+				Header:     http.Header{"Content-Type": []string{"application/json"}},
+				Body:       io.NopCloser(errorReader{}),
+			}
+		},
+		0,
+		nil,
+		nil,
+		nil,
+	},
+	{
+		"Unpack() error",
+		newTestCerviceWithNodes(),
+		&testSys,
+		[]byte("{\n  \"value\": 0,\n  \"unit\": \"\",\n  \"timestamp\": \"0001-01-01T00:00:00Z\",\n  \"version\": \"SignalA_v1.0\"\n}"),
+		func() *http.Response {
+			return &http.Response{
+				Status:     "200 OK",
+				StatusCode: 200,
+				Header:     http.Header{"Content-Type": []string{"Wrong content type"}},
+				Body:       io.NopCloser(strings.NewReader(string("{\n  \"value\": 0,\n  \"unit\": \"\",\n  \"timestamp\": \"0001-01-01T00:00:00Z\",\n  \"version\": \"SignalA_v1.0\"\n}"))),
+			}
+		},
+		0,
+		nil,
+		nil,
+		nil,
+	},
+	{
+		"DefaultClient.Do() error",
+		newTestCerviceWithNodes(),
+		&testSys,
+		[]byte("{\n  \"value\": 0,\n  \"unit\": \"\",\n  \"timestamp\": \"0001-01-01T00:00:00Z\",\n  \"version\": \"SignalA_v1.0\"\n}"),
+		func() *http.Response {
+			return &http.Response{
+				Status:     "200 OK",
+				StatusCode: 200,
+				Header:     http.Header{"Content-Type": []string{"application/json"}},
+				Body:       io.NopCloser(strings.NewReader(string("{\n  \"value\": 0,\n  \"unit\": \"\",\n  \"timestamp\": \"0001-01-01T00:00:00Z\",\n  \"version\": \"SignalA_v1.0\"\n}"))),
+			}
+		},
+		1,
+		errHTTP,
+		nil,
+		errHTTP,
+	},
+}
+
+func TestGetState(t *testing.T) {
+	for _, test := range testStateParams {
+		newMockTransport(test.body, test.mockTransportErr, test.errHTTP)
```

Comment:
Move this to line 244 (before `GetState(..)`), so the "No errors SetState" skipping occurs earlier (minimising iteration time).